sympy/functions/elementary/piecewise.py

```diff
@@ -183,25 +183,20 @@ def eval(cls, *_args):
                         free = funcs
                     _c = c
                     x = free.pop()
-                    try:
-                        c = c.as_set().as_relational(x)
-                    except NotImplementedError:
-                        pass
-                    else:
-                        reps = {}
-                        for i in c.atoms(Relational):
-                            ic = i.canonical
-                            if ic.rhs in (S.Infinity, S.NegativeInfinity):
-                                if not _c.has(ic.rhs):
-                                    # don't accept introduction of
-                                    # new Relationals with +/-oo
-                                    reps[i] = S.true
-                                elif ('=' not in ic.rel_op and
-                                        c.xreplace({x: i.rhs}) !=
-                                        _c.xreplace({x: i.rhs})):
-                                    reps[i] = Relational(
-                                        i.lhs, i.rhs, i.rel_op + '=')
-                        c = c.xreplace(reps)
+                    reps = {}
+                    for i in c.atoms(Relational):
+                        ic = i.canonical
+                        if ic.rhs in (S.Infinity, S.NegativeInfinity):
+                            if not _c.has(ic.rhs):
+                                # don't accept introduction of
+                                # new Relationals with +/-oo
+                                reps[i] = S.true
+                            elif ('=' not in ic.rel_op and
+                                    c.xreplace({x: i.rhs}) !=
+                                    _c.xreplace({x: i.rhs})):
+                                reps[i] = Relational(
+                                    i.lhs, i.rhs, i.rel_op + '=')
+                    c = c.xreplace(reps)
```

Comment:
What exactly does this do? Can it just be removed entirely?

Ideally the Piecewise constructor would not modify the args at all apart from trivial things like picking up on a True / False condition.

Reply:
It's been a long time since I had a look at it and I do not think I ever read this part (focused more on the simplification). But I can give it a try and see.

Reply:
The piecewise tests pass without it, so pushing to see the whole test suite.